forms-flow-components/src/components/CustomComponents/DateFilter.tsx

```diff
@@ -78,55 +91,87 @@ export const DateRangePicker: FC<DateRangePickerProps> = ({
   useEffect(() => {
-    if (initialDateRange) {
+    if (value) {
       const newDateRange = {
-        startDate: initialDateRange.startDate
-          ? parseDate(initialDateRange.startDate)
-          : new Date(),
-        endDate: initialDateRange.endDate
-          ? parseDate(initialDateRange.endDate)
-          : new Date(new Date().setDate(new Date().getDate() + 30)),
+        startDate: value?.startDate ? parseDate(value.startDate) : null,
+        endDate: value?.endDate ? parseDate(value.endDate) : null,
       };
+
       setDateRange(newDateRange);
 
-      // Safely update current month
+      // Safely update current month if startDate exists
       if (newDateRange.startDate) {
         setCurrentMonth(
           newDateRange.startDate instanceof Date
-            ? newDateRange.startDate
+            ? new Date(newDateRange.startDate)
             : parseDate(newDateRange.startDate)
         );
       }
+    } else {
+      // Reset to empty range if value is undefined/null
+      setDateRange(getDefaultDateRange());
     }
-  }, [initialDateRange]);
+  }, [value]);
 
   // Format date according to specified format (default: MM/DD/YYYY)
   const formatDate = (date: Date | null): string => {
-    if (!date) return "MM/DD/YYYY";
+    if (!date) {
+      // Return a properly formatted placeholder using the dateFormat pattern
+      return dateFormat
+        .replace(/M+/g, "MM")
+        .replace(/D+/g, "DD")
+        .replace(/Y+/g, "YYYY");
+    }
 
     // Default format (MM/DD/YYYY)
     const month = String(date.getMonth() + 1).padStart(2, "0");
     const day = String(date.getDate()).padStart(2, "0");
     const year = date.getFullYear();
-    return `${month}/${day}/${year}`;
+
+    // Apply the specified format
+    let formattedDate = dateFormat;
+    formattedDate = formattedDate.replace(/M+/g, month);
+    formattedDate = formattedDate.replace(/D+/g, day);
+    formattedDate = formattedDate.replace(/Y+/g, year.toString());
+
+    return formattedDate;
   };
 
   const getFormattedDate = (date: Date | string | null): string => {
-    if (!date) return "MM/DD/YYYY";
+    if (!date) {
+      // Return a properly formatted placeholder using the dateFormat pattern
+      // This ensures we return "MM/DD/YYYY" instead of repeating characters
+      return dateFormat
+        .replace(/M+/g, "MM")
+        .replace(/D+/g, "DD")
+        .replace(/Y+/g, "YYYY");
+    }
     const parsedDate = date instanceof Date ? date : parseDate(date);
     return formatDate(parsedDate);
   };
 
   const formatDateRange = (): string => {
-    const start = getFormattedDate(dateRange?.startDate ?? null);
-    const end = getFormattedDate(dateRange?.endDate ?? null);
+    if (!dateRange.startDate && !dateRange.endDate) {
+      return t(placeholder); // Added translation tag
+    }
+
+    // Modified to show date format instead of dash for missing dates
+    const start = dateRange?.startDate
+      ? getFormattedDate(dateRange.startDate)
```

Comment:
can we optimize this call like this right getFormattedDate(dateRange.startDate) because if the startDate is there it will return parsed date else it will return this code right 
`  return dateFormat
        .replace(/M+/g, "MM")
        .replace(/D+/g, "DD")
        .replace(/Y+/g, "YYYY");`